server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/ClusterBalanceStats.java

```diff
@@ -231,7 +231,7 @@ private static NodeBalanceStats createFrom(
             long actualShardSize = 0L;
 
             for (ShardRouting shardRouting : routingNode) {
-                var indexMetadata = metadata.getProject().index(shardRouting.index());
+                var indexMetadata = metadata.indexMetadata(shardRouting.index());
                 var shardSize = clusterInfo.getShardSize(shardRouting, 0L);
                 assert indexMetadata != null;
```

Comment:
Nit: in this case, I'd argue the assertion loses its value because a non-existent index (or project) would have already thrown an exception.